graphql/graphql.go

```diff
@@ -478,8 +478,11 @@ func (t *Transaction) getLogs(ctx context.Context) (*[]*Log, error) {
 	if err != nil {
 		return nil, err
 	}
-	ret := make([]*Log, 0, len(logs))
+	ret := make([]*Log, 0)
 	for _, log := range logs {
+		if uint64(log.TxIndex) != t.index {
+			continue
```

Comment:
`logs` is sorted by TxIndex, so this could be improved to skip most iteration:

```
ix := sort.Search(len(logs), func (i int) bool { return logs[i].TxIndex == t.index })
for ix < len(logs) && logs[ix].TxIndex == t.index {
    ret = append(ret, logs[i])
}
```

Reply:
Also, it's kind of weird to use a filter here. We could just call `t.r.backend.GetLogs` here.

Reply:
Hmm, just realized that the filter is used here exactly because `GetLogs` doesn't have access to the cache.

Reply:
I kind of like the simplicity even tho it's slightly less fast, if you don't have a strong feeling about it.

Reply:
Fix tested working on my personal node.
Would deploy 1.10.23 + your_fix on prod if no 1.10.24 official version before the merge (currently using 1.10.21 + TTD set in CLI params)

Reply:
I'm slightly in favor of @fjl's suggestion.